projects/core/src/lib/components/legend/services/legend.service.ts

```diff
@@ -63,6 +63,69 @@ export class LegendService {
       );
   }
 
+  public getLegendImages$(appLayers$: Observable<AppLayerWithServiceModel[]>, urlCallback?: (layer: AppLayerWithServiceModel, url: URL) => void):
+    Observable<Array<{ appLayer: AppLayerWithServiceModel; imageData: string; width: number; height: number }>> {
+    return this.getAppLayerAndUrl$(appLayers$).pipe(
+      concatMap(appLayerAndUrls => {
+        if (appLayerAndUrls.length === 0) {
+          return of([]);
+        }
+        return forkJoin(appLayerAndUrls.filter(lu => lu.url !== '').map(appLayerWithLegendUrl => {
+          const url = new URL(appLayerWithLegendUrl.url);
+          if (urlCallback) {
+            urlCallback(appLayerWithLegendUrl.layer, url);
+          }
+          return LegendService.imageUrlToPng$(url.toString()).pipe(
+            catchError((e) => {
+              console.log(`Error getting legend from URL ${appLayerWithLegendUrl.url}`, e);
+              return of(null);
+            }),
+            map(legendImage => ({
```

Comment:
If I read correctly this legendImage can be `null`. Not sure what

```{ ...null, appLayer: ... }``` will do but maybe its good to check if legendImage is not null. If its null just return null here, you filter this later.